src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp

```diff
@@ -518,8 +514,8 @@ CHIP_ERROR Spake2pHSM_P256_SHA256_HKDF_HMAC::KeyConfirm(const uint8_t * in, size
     {
         return Spake2p::KeyConfirm(in, in_len);
     }
-
-    const CHIP_ERROR error = Spake2p_KeyConfirm_HSM(&hsm_pake_context, role, in, in_len);
+    CHIP_ERROR error = CHIP_ERROR_INTERNAL;
+    error            = Spake2p_KeyConfirm_HSM(&hsm_pake_context, role, in, in_len);
```

Comment:
Why is this change needed?

Reply:
And in particular, why can't `error` stay const here?

Reply:
It is the same now.. After the latest commit. This had got changed for VerifyOrExit which was being used sometime earlier and "error variable" was being reused. now VerifyOrReturnError is being used and the change is not required.